File: src/main/java/kr/co/lunatalk/global/security/PrincipalDetails.java
package kr.co.lunatalk.global.security;

import kr.co.lunatalk.domain.member.domain.MemberRole;
import lombok.AllArgsConstructor;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.Collections;
import java.util.List;

@AllArgsConstructor
public class PrincipalDetails implements UserDetails {
	private final Long memberId;
	private final MemberRole memberRole;


	@Override
	public Collection<? extends GrantedAuthority> getAuthorities() {
		return Collections.singleton(new SimpleGrantedAuthority(memberRole.getValue()));
	}

	@Override
	public String getPassword() {
		return null;
	}

	@Override
	public String getUsername() {
		return memberId.toString();
	}

	@Override
	public boolean isAccountNonExpired() {
		return true;
	}

	@Override
	public boolean isAccountNonLocked() {
		return true;
	}

	@Override
	public boolean isCredentialsNonExpired() {
		return true;
	}

	@Override
	public boolean isEnabled() {
		return true;
	}
}

File: src/main/kotlin/kr/co/lunatalk/global/security/PrincipalDetails.kt
package kr.co.lunatalk.global.security

import kr.co.lunatalk.domain.member.domain.MemberRole
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

class PrincipalDetails(
    private val memberId: Long,
    private val memberRole: MemberRole
) : UserDetails {

    override fun getAuthorities(): Collection<GrantedAuthority> =
        setOf(SimpleGrantedAuthority(memberRole.value))

    override fun getPassword(): String? = null

    override fun getUsername(): String = memberId.toString()

    override fun isAccountNonExpired(): Boolean = true

    override fun isAccountNonLocked(): Boolean = true

    override fun isCredentialsNonExpired(): Boolean = true

    override fun isEnabled(): Boolean = true
}
